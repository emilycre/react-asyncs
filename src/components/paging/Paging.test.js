import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging Component', () => {
  it('Renders Paging', () => {
    const wrapper = shallow(
      <Paging currentPage={1} totalPages={10} nextPage={jest.fn()}>
        <h1>Hello.</h1>
      </Paging>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Increments page on Next Page button click', () => {
    const next = jest.fn();
    const wrapper = shallow(
      <Paging currentPage={2} totalPages={10} nextPage={next}>
        <h1>Hello, again.</h1>
      </Paging>
    );
    wrapper.find('button').at(1).simulate('click');
    expect(next).toHaveBeenCalledWith(3);
  });

  
});
