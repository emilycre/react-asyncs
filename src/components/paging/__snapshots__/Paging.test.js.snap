// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging Component Renders Paging 1`] = `
<Fragment>
  1 / 10
  <button
    onClick={[Function]}
  >
    Next Page
  </button>
  <h1>
    Hello.
  </h1>
</Fragment>
`;
